dask/array/tests/test_routines.py

```diff
@@ -22,6 +22,51 @@ def test_array():
     assert isinstance(y, da.Array)
 
 
+@pytest.mark.parametrize("funcname", [
+    "atleast_1d",
+    "atleast_2d",
+    "atleast_3d",
+])
+@pytest.mark.parametrize("num_arrs", [
+    0,
+    1,
+    2
+])
+@pytest.mark.parametrize("shape, chunks", [
+    (tuple(), tuple()),
+    ((4,), (2,)),
+    ((4, 6), (2, 3)),
+    ((4, 6, 8), (2, 3, 4)),
+    ((4, 6, 8, 10), (2, 3, 4, 5)),
+])
+def test_atleast_nd(funcname, num_arrs, shape, chunks):
```

Comment:
This adds 45 tests for just these 3 functions. Any way you can simplify this a bit? I might move the checks for different number of arrays inside the test, as you don't really need to check each bit for 0 arrays or 1 array.

```python
assert np_func() == da_func()  # All the testing needed for 0 arrays

assert_eq(da_func(da_a), np_func(np_a))
assert_eq(da_func(np_a), np_func(np_a))  # Check with non-dask array

# checks for multiple arrays...
```

Reply:
Ok let me give that some thought.